atef/config.py

```diff
@@ -179,8 +304,37 @@ def compare(self) -> Result:
                     f"for comparison {self.comparison}"
                 ),
             )
-        self.result = self.comparison.compare_signal(
-            self.signal,
+
+        try:
+            self.data = await self.get_data()
```

Comment:
I'm surprised that this bit and the corresponding tool runs aren't done when we prepare the comparison. Maybe I misunderstand the staging here about what is expected in the prepare step vs the compare step. I'm slightly concerned about this mainly for the dynamic configurations where we might get each side of the comparison inequality at different points in time.

Reply:
I think you're right and this needs some work, similar to the above.